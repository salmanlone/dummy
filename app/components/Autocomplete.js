const linkStyle = {
  margin: "0px 15px"
};

const Autocomplete = props => (
  <input style={linkStyle} type="text" placeholder={props.placeholder} />
);

export default Autocomplete;
