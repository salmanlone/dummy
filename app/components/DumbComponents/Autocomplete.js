const linkStyle = {
  margin: "0px 15px"
};

const Autocomplete = props => (
  <input style={linkStyle} type="text" placeholder={props.placeholder} onChange={e=> props.callbackHandler(e)} />
);

export default Autocomplete;
