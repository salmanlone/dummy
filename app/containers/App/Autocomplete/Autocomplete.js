import React, { useState, useEffect } from "react";
import renderHTML from "react-render-html";
import styles from './Autocomplete.module.css';


const Autocomplete = ({ placeholder, callbackHandler, DataFromState  }) => {

  const [activeSuggestion, setActiveSuggestion] = useState(0);
  const [filteredSuggestions, setFilteredSuggestions] = useState([]);
  const [showSuggestions, setShowSuggestions] = useState(false);
  const [userInput, SetUserInput] = useState('');


  useEffect(() => {
    setFilteredSuggestions(DataFromState);
    setActiveSuggestion(0);
    setShowSuggestions(true);
  }, [DataFromState]);

  const onChange = (e) => {
    SetUserInput(e.target.value);
  }

  const onMouseDown = (e) => {
    setActiveSuggestion(0);
    setFilteredSuggestions([]);
    setShowSuggestions(false);
    SetUserInput(e.currentTarget.innerText)
  };

  const onKeyDown = (e) => {

    if (e.keyCode === 13) {
      setActiveSuggestion(0);
      setShowSuggestions(false);
      SetUserInput(filteredSuggestions[activeSuggestion].Text.replace(
        "<b>",
        ""
      ).replace("</b>", ""))

    }
    else if (e.keyCode === 38) {
      if (activeSuggestion === 0) {
        return;
      }

      setActiveSuggestion(activeSuggestion - 1)

    }
    else if (e.keyCode === 40) {
      if (activeSuggestion - 1 === filteredSuggestions.length) {
        return;
      }
      setActiveSuggestion(activeSuggestion + 1)
    }
  };

  const onBlur = (e) => {
    setActiveSuggestion(0);
    setShowSuggestions(false);
  };


  let suggestionsListComponent;

  if (showSuggestions) {
    if (filteredSuggestions) {
      suggestionsListComponent = (
        <ul className={styles.suggestions} >
          {filteredSuggestions.map((suggestion, index) => {

            return (
              <li
                className= { index === activeSuggestion ? styles.suggestion_active : ""}
                key={suggestion.ID}
                onMouseDown={onMouseDown}
              >
                {renderHTML(suggestion.Text)}
              </li>
            );
          })}
        </ul>
      );
    } else {
      suggestionsListComponent = (
        <div className={styles.no_suggestions} >
          {/* <em>No suggestions</em> */}
        </div>
      );
    }
  }




  return (
    <React.Fragment>
      <input className={styles.input}
        type="text"
        placeholder={placeholder}
        onChange={e => { onChange(e), callbackHandler(e) }}
        onKeyDown={onKeyDown}
        onBlur={onBlur}
        value={userInput}
      />
      {suggestionsListComponent}
    </React.Fragment>
  )
};



export default Autocomplete;
