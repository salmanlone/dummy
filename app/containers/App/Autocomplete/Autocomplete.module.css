
.no_suggestions {
    color: #999;
    padding: 0.5rem;
    margin: 0px 15px;
  }
  
  .suggestions {
    border: 1px solid #999;
    border-top-width: 0;
    list-style: none;
    margin-top: 0;
    max-height: 143px;
    overflow-y: auto;
    padding-left: 0;
    width: calc(300px + 1rem);
    margin: 0px 15px;
  }
  
  .suggestions li {
    padding: 0.5rem;
  }
  
  .suggestions li:hover {
    background-color: #c0c9c6;
    color: #080808;
    cursor: pointer;
    font-weight: 700;
  }

  .suggestion_active {
    background-color: #c0c9c6;
    color: #080808;
    cursor: pointer;
    font-weight: 700;
  }
  
  .suggestions li:not(:last-of-type) {
    border-bottom: 1px solid #999;
  }

  .input{
    margin: 0px 15px;
    border: 1px solid #999;
    padding: 0.5rem;
    width: 300px;
  }