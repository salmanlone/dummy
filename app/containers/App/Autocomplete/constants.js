export const SET_USER_INPUT = "SET_USER_INPUT";
