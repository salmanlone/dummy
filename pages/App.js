class App extends React.Component {
    render() {
        return (
            <div>Hello, Sunshine!</div>
        );
    }
}

export default App;