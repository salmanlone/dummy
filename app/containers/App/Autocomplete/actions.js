import * as actionTypes from "./constants";

export function setStateUserInput(userInput, name) {
  return {
    type: actionTypes.SET_USER_INPUT,
    userInput,
    name
  };
}
