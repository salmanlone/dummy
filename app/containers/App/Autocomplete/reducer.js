import * as actionTypes from "./constants";

export const initialState = {
  selectedLocation: "",
  selectedPosition: ""
};

const AutocompleteReducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.SET_USER_INPUT:
      switch (action.name) {
        case "location":
          return {
            ...state,
            ...{ selectedLocation: action.userInput }
          };
        case "position":
          return {
            ...state,
            ...{ selectedPosition: action.userInput }
          };
        default:
          return state;
      }
    default:
      return state;
  }
};

export default AutocompleteReducer;
