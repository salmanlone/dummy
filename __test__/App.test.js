import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import App from "../pages/App.js";
import Autocomplete from "../app/components/Autocomplete";

describe("With Enzyme", () => {
  it('App shows "Hello, Sunshine!"', () => {
    const app = shallow(<App />);

    expect(app.find("div").text()).toEqual("Hello, Sunshine!");
  });
});

describe("With Enzyme Home", () => {
    it('App shows "Hello, Home!"', () => {
      const autocomplete = shallow(<Autocomplete placeholder="test"/>);
  
      expect(autocomplete.find("input").props().placeholder).toEqual("test");
    });
  });

describe("With Snapshot Testing", () => {
  it('App shows "Hello, Sunshine!"', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});