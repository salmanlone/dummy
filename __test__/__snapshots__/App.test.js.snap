// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello, Sunshine!" 1`] = `
<div>
  Hello, Sunshine!
</div>
`;
